Periodically-Syncing-HyperChains.md

```diff
@@ -533,12 +533,78 @@ You can not withdraw funds if the deposited balance would go under `MINIMUM_STAK
 if you have tokens at stake.
 
 #### Staking Cycle Structure
-A staking cycle consists of four distinct epochs:
 
-1. **Staking Epoch**: Participants register and adjust their stakes.
-2. **Leader Election Epoch**: The system uses the state of the parent chain and the stakes recorded to generate a schedule for selecting leaders.
-3. **Block Production Epoch**: Only validators meeting the minimum staking requirements are eligible to produce blocks.
-4. **Payout Epoch**: Rewards are distributed based on block production results.
+Each epoch needs a schedule of producers. These producers are randomly selected from registered stakers.
+We are concerned about the following:
+- To prevent stakers to influence the schedule, the staking distribution should be known before the seed of the random selection is known.
+- The seed should be final (that is it should be guaranteed not to disappear from the parent chain due to forking)
+- The seed should be known before the actual schedule needs to be computed, otherwise the child chain is stuck
+- The stakers should actually do work, so only after the work is done, their rewards for doing the work is paid out.
+
+This leads to a design with of a staking cycle that consists of four distinct phases:
+
+1. **Staking Phase**: Participants register and adjust their stakes.
+   In the staking phase, we collect all stakes posted in the ongoing child epoch. The result at the Nth child epoch CE(n) is refered to as s(n).  The initial stake as configured for the child chain is s(0).
+
+2. **Leader Election Phase**: The system uses the state of the parent chain and the stakes recorded to generate a schedule for selecting leaders.
+   In the leader election phase we retrieve the last hash of a parent chain epoch to be used as a seed later on. For CE(n) we store the last blockhash of PE(n-2) as seed for later schedule computation.
```

Comment:
I think currently we use the _first_ block of parent epochs, but that is just a technicality.

Reply:
Good point.